lib/classes/ConfigSchemaHandler/errorMessageBuilder.js

```diff
@@ -0,0 +1,171 @@
+      }
+      return `${ajvError.dataPath.slice(1)} ${ajvError.message} (received '${additionalProperty}')`;
+    }
+    default:
+      return `${ajvError.dataPath.slice(1)} ${ajvError.message}`;
+  }
+};
+
+const addUserFriendlyMessage = (errors, userConfig) => {
+  return errors.map(error => {
+    let friendlyMessage;
+
+    // regex matches '.functions['xxx'].events[x].xxx'
+    const isEventParam = /^\.functions\['[\w-_]+'\]\.events\[[0-9]+\]\.[\w]+$/.test(error.dataPath);
+    if (error.params && error.params.additionalProperty && isEventParam) {
+      const eventName = error.dataPath.split('.')[3];
+      friendlyMessage = `Unsupported parameter '${error.params.additionalProperty}' for '${eventName}' event`;
+    }
+
+    // regex matches '.functions['xxx'].events[x]'
+    const isFunctionEvent = /^\.functions\['[\w-_]+'\]\.events\[[0-9]+\]$/.test(error.dataPath);
+    if (error.params && error.params.additionalProperty && isFunctionEvent) {
+      friendlyMessage = `Unsupported function event '${
+        error.params.additionalProperty
+      }' at ${error.dataPath.slice(1)}`;
+    }
+
+    const isEventAnyOfErrorObject =
+      error.keyword === 'anyOf' &&
+      error.schemaPath === ANY_OF_EVENT_SCHEMA_PATH &&
+      error.params.suberrors &&
+      error.params.suberrors.length &&
+      error.params.suberrors[0].params &&
+      error.params.suberrors[0].params.additionalProperty;
+
+    if (isEventAnyOfErrorObject) {
+      friendlyMessage = `Unsupported function event '${
+        error.params.suberrors[0].params.additionalProperty
+      }' at ${error.dataPath.slice(1)}`;
+    }
+
+    const configUnderValidation = _.get(userConfig, error.dataPath.slice(1));
+
+    if (isEventAnyOfErrorObject && Object.keys(configUnderValidation).length > 1) {
+      const contextStr = `${Object.keys(configUnderValidation).length} (${Object.keys(
+        configUnderValidation
+      ).join(', ')})`;
+      friendlyMessage = [
+        'Event should contain only one root property,',
+        `but got ${contextStr} at ${error.dataPath.slice(1)}`,
+      ].join(' ');
+    }
+
+    if (error.dataPath === '.package' && error.params && error.params.additionalProperty) {
+      friendlyMessage = [
+        `Unsupported package property '${error.params.additionalProperty}'.`,
+        'Avaiable properties: individually, path, artifact, exclude, include, excludeDevDependencies.',
+      ].join(' ');
+    }
+
+    if (error.dataPath === '.functions' && error.params && error.params.additionalProperty) {
+      friendlyMessage = `Function name '${
+        error.params.additionalProperty
+      }' must be alphanumeric at ${error.dataPath.slice(1)}.`;
+    }
+
+    // This is the case when error is related to root level of config
+    if (error.dataPath === '' && error.params && error.params.additionalProperty) {
+      friendlyMessage = `Unsupported root property '${error.params.additionalProperty}'`;
```

Comment:
I guess this one can be handled with same patch proposed above, just host name should be `root`

Reply:
Adjusted to use a general case